tests/testthat/test-Samples-methods.R

```diff
@@ -23,3 +25,135 @@ test_that("names-Samples returns correct names of the parameters", {
+    rng_seed = 544914,
+    rng_kind = "Mersenne-Twister"
+  )
+
+  samples <- mcmc(data, model, options)
+
+  expect_error(
+    approximate(
+      object = samples,
+      model = model,
+      data = data,
+      logNormal = "NotLogical",
+      control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+    ),
+    "Assertion on 'logNormal' failed: Must be of type 'logical', not 'character'"
+  )
+  expect_error(
+    approximate(
+      object = samples,
+      model = model,
+      data = data,
+      verbose = "NotLogical",
+      control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+    ),
+    "Assertion on 'verbose' failed: Must be of type 'logical', not 'character'"
+  )
+  expect_error(
+    approximate(
+      object = samples,
+      model = model,
+      data = data,
+      create_plot = "NotLogical",
+      control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+    ),
+    "Assertion on 'create_plot' failed: Must be of type 'logical', not 'character'"
+  )
+  expect_error(
+    approximate(
+      object = samples,
+      model = model,
+      data = data,
+      refDose = "NotNumeric",
+      control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+    ),
+    "Assertion on 'refDose' failed: Must be of type 'numeric', not 'character'"
+  )
+  expect_error(
+    approximate(
+      object = samples,
+      model = model,
+      data = data,
+      points = c(1:5, "NotNumeric"),
+      control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+    ),
+    "Assertion on 'points' failed: Must be of type 'numeric', not 'character'"
+  )
+})
+
+test_that("Approximate works correctly", {
+  data <- Data(
+    x = c(0.1, 0.5, 1.5, 3, 6, 10, 10, 10),
+    y = c(0, 0, 0, 0, 0, 0, 1, 0),
+    ID = 1L:8L,
+    cohort = c(0, 1, 2, 3, 4, 5, 5, 5),
+    doseGrid = c(0.1, 0.5, 1.5, 3, 6, seq(from = 10, to = 80, by = 2))
+  )
+
+  model <- LogisticLogNormal(
+    mean = c(-0.85, 1),
+    cov = matrix(c(1, -0.5, -0.5, 1), nrow = 2),
+    ref_dose = 56
+  )
+
+  options <- McmcOptions(
+    burnin = 100,
+    step = 2,
+    samples = 2000,
+    rng_seed = 544914,
+    rng_kind = "Mersenne-Twister"
+  )
+
+  samples <- mcmc(data, model, options)
+
+  actual <- approximate(
+    object = samples,
+    model = model,
+    data = data,
+    control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+  )
+
+  expect_equal(length(actual), 2)
+  expect_set_equal(names(actual), c("model", "plot"))
+  for (slot_name in slotNames(actual$model)) {
+    if (!is.function(slot(actual$model, slot_name))) {
+      expect_snapshot(slot(actual$model, slot_name))
+    }
+  }
+  vdiffr::expect_doppelganger("approximate-Samples", actual$plot)
+
+  actual1 <- approximate(
+    object = samples,
+    model = model,
+    data = data,
+    create_plot = FALSE,
+    control = list(threshold.stop = 0.1, max.time = 1, maxit = 1)
+  )
+  expect_equal(length(actual1), 1)
+  expect_set_equal(names(actual1), c("model"))
+})
+
+## nolint end
```

Comment:
since we have cleaned up the tests, please avoid nolint start and end. All of this should be checked by the linter

Reply:
Done.